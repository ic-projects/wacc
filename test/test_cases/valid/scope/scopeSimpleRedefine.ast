-- Test: scopeSimpleRedefine.wacc

-- Uploaded file: 
---------------------------------------------------------------
# variable scoping test that redefines a variable

# Output:
# true
# 12

# Program:

begin
  int x = 12 ;
  begin
    bool x = true ;
    println x
  end ;
  println x 
end

---------------------------------------------------------------

-- Compiler Output:
-- Compiling...
-- Printing AST...
scopeSimpleRedefine.ast contents are:
===========================================================
0	Program
1	  - int main()
2	    - DECLARE
3	      - TYPE
4	        - int
5	      - LHS
6	        - x
7	      - RHS
8	        - 12
9	    - SCOPE
10	      - DECLARE
11	        - TYPE
12	          - bool
13	        - LHS
14	          - x
15	        - RHS
16	          - true
17	      - PRINTLN
18	        - x
19	    - PRINTLN
20	      - x
===========================================================
-- Finished

