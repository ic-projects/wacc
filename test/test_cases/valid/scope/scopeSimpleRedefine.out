-- Compiling...
-- Assembling and Linking...
-- Executing...
===========================================================
true
12
===========================================================
The exit code is 0.
-- Finished
